### PR TITLE
Add exception chaining

### PR DESCRIPTION
**Pull request by DanielNoord**
Closes https://github.com/channable/opnieuw/issues/22.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,7 +16,10 @@ Release highlights:
 
 - Remove `DoCall`, `DoWait` and `RetryState`.
 - Use `BackoffCalculator` to keep track of the time to wait between retries and refactor
-  retrying logic accordingly.
+  retrying logic accordingly.0
```
Comment (jochemb) [inline CHANGELOG.md:19]:
```suggestion
  retrying logic accordingly.
```

**Review by jochemb — APPROVED** [review]
One typo, other than that LGTM

**Comment by DanielNoord** [comment]
@OpsBotPrime merge on friday

**Comment by OpsBotPrime** [comment]
Pull request approved for merge by @DanielNoord, rebasing now.

**Comment by OpsBotPrime** [comment]
Rebased as bd0e7ca703716d8e53b066ca99962728445247a8, waiting for CI …

**Comment by OpsBotPrime** [comment]
[CI job :yellow_circle:](https://channable.semaphoreci.com/workflows/d1db7c49-a7fd-4bdc-a1c9-27b9117b0d3b?pipeline_id=0e41e3de-7aef-4913-88ab-56bd1f2b7854) started.

**Pull request merged**